src/ace/managers/viewport/tilebuffer.c

```diff
@@ -299,17 +413,44 @@ void tileBufferProcess(tTileBufferManager *pManager) {
 				uwTileOffsY = (pState->pMarginX->wTileCurr << ubTileShift) & (pManager->uwMarginedHeight-1);
 				uwTileOffsX = (pState->pMarginX->wTilePos << ubTileShift);
 				// Redraw remaining tiles
-				while (pState->pMarginX->wTileCurr < pState->pMarginX->wTileEnd) {
-					tileBufferDrawTileQuick(
-						pManager, pState->pMarginX->wTilePos, pState->pMarginX->wTileCurr,
-						uwTileOffsX, uwTileOffsY
+				UWORD uwBltsize = tileBufferSetupTileDraw(pManager);
+				UWORD uwTileCurr = pState->pMarginX->wTileCurr;
+				UWORD uwTileEnd = pState->pMarginX->wTileEnd;
+				UWORD uwMarginedHeight = pManager->uwMarginedHeight;
+				UWORD uwTilePos = pState->pMarginX->wTilePos;
+				UBYTE *pTileColumn = pManager->pTileData[uwTilePos];
+				UWORD uwDstBytesPerRow = pManager->pScroll->pBack->BytesPerRow;
+				PLANEPTR pDstPlane = pManager->pScroll->pBack->Planes[0];
+				ULONG ulDstOffs = uwDstBytesPerRow * uwTileOffsY + (uwTileOffsX >> 3);
+				UWORD uwDstOffsStep = uwDstBytesPerRow * ubTileSize;
+				// set up the first bltdpt for an interleaved blit. if this isn't
+				// interleaved, this is wasted, but interleaved will be faster with
+				// this. we can just do this here, since tileBufferSetupTileDraw
+				// already waited for the blitter to be idle
+				g_pCustom->bltdpt = pDstPlane + ulDstOffs;
+				while (uwTileCurr < uwTileEnd) {
+					tileBufferContinueTileDraw(
+						pManager, pTileColumn, uwTileCurr,
+						uwBltsize, ulDstOffs, pDstPlane,
+						// do not set bltdpt, it was left at the right place by the previous blit
+						0
 					);
-					++pState->pMarginX->wTileCurr;
-					uwTileOffsY = (uwTileOffsY + ubTileSize);
-					if(uwTileOffsY >= pManager->uwMarginedHeight) {
-						uwTileOffsY -= pManager->uwMarginedHeight;
+					if (cbTileDraw) {
+						cbTileDraw(uwTilePos, uwTileCurr, pManager->pScroll->pBack, uwTileOffsX, uwTileOffsY);
```

Comment:
it occured to me that this can't be called here if we expect end users to draw stuff on those tiles - calling any other blit function will trash the blitter registers. So this needs to be in a separate loop.

The good news is, it's a speedup for non-callback mode - it can only check if cbTileDraw is non-zero only once, and if so, iterate over new tiles.